lib/utils/docs_parser/output_formatters/base_formatter.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:solid_lints/utils/docs_parser/models/rule_doc.dart';
+
+/// Base class for output formatter
+abstract class BaseFormatter {
+  /// List of rules to format
+  final List<RuleDoc> rules;
+
+  ///
+  const BaseFormatter(this.rules);
+
+  /// Get formatted string containing documentation of given [rules]
+  String format() {
+    final formattedResult = StringBuffer();
+    for (final rule in rules) {
+      formattedResult.writeln(formatRuleDoc(rule));
+    }
+
+    return formattedResult.toString();
+  }
+
+  /// Get formatted string for the single given [rule]
+  String formatRuleDoc(RuleDoc rule);
+}
```

Comment:
Think about how this may support outputting a file per rule. For now the base class does little to support it.

Perhaps we can keep it as simple as possible. Let the concrete implementation determine what the outputs are:

```dart
interface class RulesDocumentationFormatter<T> {
  T format(List<RuleDoc> docs);
}
```